Tools/wasm/emscripten/node_entry.mjs

```diff
@@ -12,19 +11,34 @@ if (process?.versions?.node) {
   }
 }
 
+function rootDirsToMount(Module) {
+  return fs
+      .readdirSync("/")
+      .filter((dir) => !["dev", "lib", "proc"].includes(dir)).map(dir => "/" + dir);
+}
+
+function mountDirectories(Module) {
+  for (const dir of rootDirsToMount(Module)) {
+    Module.FS.mkdirTree(dir);
+    Module.FS.mount(Module.FS.filesystems.NODEFS, { root: dir }, dir);
+  }
+}
+
+const pythonShIndex = process.argv.findIndex(x => x.endsWith("python.sh"));
+
 const settings = {
   preRun(Module) {
-    const __dirname = dirname(fileURLToPath(import.meta.url));
-    Module.FS.mkdirTree("/lib/");
-    Module.FS.mount(Module.FS.filesystems.NODEFS, { root: __dirname + "/lib/" }, "/lib/");
+    Module.FS.mkdirTree("/home/");
+    mountDirectories(Module);
+    Module.FS.chdir(process.cwd());
+    Object.assign(Module.ENV, process.env);
```

Comment:
Related: Does this change require any update to the docs that landed in #127113 to reference `/home/lib`, rather than `/lib`? I guess there's nothing that *requires* the use of the `/home` prefix, but for consistency/establishing best practice, would it be advisable?

Reply:
Yeah, what I'm doing here is mounting as much as possible of your native file system. I dropped `/dev`, `/proc`, and `/lib` because they contain native devices, processes, and libraries whereas we want to tell people about the Emscripten devices and processes. Though honestly unlinking `/dev` and bringing in the native `/dev` would probably not be the worst idea. Something to think about.

Reply:
I don't think any documentation update is needed. As long as `configure.ac` sets `host_prefix` to `/lib`, mounting the standard library there will allow it to be located. I don't think it would work to mount it to `/home/lib`...